tiny_dnn/core/framework/tensor.h

```diff
@@ -147,59 +147,88 @@ class Tensor {
     }
 #endif
 
-    // Returns the tensor shape
-    const std::array<size_t, 4>& shape() const { return shape_; }
-
-    // Returns the value of a specified index in the tensor.
-    // Checked version (throw exceptions for out-of-range error)
-    U& host_at(const size_t d0,
-               const size_t d1,
-               const size_t d2,
-               const size_t d3) {
-        return *host_ptr(d0, d1, d2, d3);
-    }
-
-    U host_at(const size_t d0,
-              const size_t d1,
-              const size_t d2,
-              const size_t d3) const {
-        return *host_ptr(d0, d1, d2, d3);
-    }
-
-    // Returns the pointer to a specified index in the tensor
-    // Checked version (throw exceptions for out-of-range error)
-    const U* host_ptr(const size_t d0,
-                      const size_t d1,
-                      const size_t d2,
-                      const size_t d3) const {
-        if (d0 >= shape_[0] || d1 >= shape_[1] ||
-            d2 >= shape_[2] || d3 >= shape_[3]) {
+    /**
+     *
+     * @return the tensor shape
+     */
+    const std::array<size_t, kDimensions>& shape() const { return shape_; }
+
+    /**
+     * Checked version of access to indexes in tensor (throw exceptions
+     * for out-of-range error)
+     * @param args indexes in tensor
+     * @return the value of a specified index in the tensor
+     */
+    template<typename... Args>
+    U& host_at(const Args... args) {
```

Comment:
the motivation behind this is for testing so that you can modify at runtime values